runtime/hydradx/src/lib.rs

```diff
@@ -1186,4 +1181,66 @@ impl_runtime_apis! {
 			Default::default()
 		}
 	}
+	impl sp_api::Core<Block> for Runtime {
+		fn version() -> RuntimeVersion {
+			VERSION
+		}
+
+		fn execute_block(block: Block) {
+			Executive::execute_block(block)
+		}
+
+		fn initialize_block(header: &<Block as BlockT>::Header) -> ExtrinsicInclusionMode {
+			Executive::initialize_block(header)
+		}
+	}
+	impl primitives::runtime_api::FeeEstimationApi<Block, AccountId, AssetId, Balance> for Runtime {
+		fn estimate_fee_payment(weight: Weight, account_id: AccountId) -> (AssetId, Balance) {
+			// Get the currency configured for this account
+			let currency = MultiTransactionPayment::account_currency(&account_id);
+
+			// Convert weight to base fee using native asset calculations
+			let native_fee = TransactionPayment::weight_to_fee(weight);
+
+			// Get the price of the account's currency if applicable
+			let price = MultiTransactionPayment::price(currency).unwrap_or(Price::one());
```

Comment:
this is not correct. if there is not price, it should not return anything, hence the comment about return an option.